### PR TITLE
redo(ticdc): fix redo initialization block the owner

### LINKED ISSUE #9886: redo read metadata may block the owner and read bad data
### What did you do?

1. mount a NFS as the external storage for the redo log.
2. create a changefeed with the redo enabled.

### What did you expect to see?

the changefeed works normally

### What did you see instead?

* the changefeed cannot contact the NFS system.
* the owner is blocked for a long time, around 8 minutes.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
v6.5.5
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9886

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
none
```

### CODE REVIEW

File: cdc/redo/manager.go [hunk 1]
```diff
@@ -263,21 +251,30 @@ func newLogManager(
...
-		return m.bgUpdateLog(ctx)
+	failpoint.Inject("ChangefeedNewRedoManagerError", func() {
+		failpoint.Return(errors.New("changefeed new redo manager injected error"))
+	})
+	if !m.Enabled() {
+		return nil
 	}
-	return nil
+
+	defer m.close()
```
Comment (CharlesCheung96) [inline cdc/redo/manager.go:265]: m.writer may be nil when close is called.

File: cdc/redo/meta_manager.go [hunk 2]
```diff
@@ -46,13 +47,19 @@ type MetaManager interface {
 	// Cleanup deletes all redo logs, which are only called from the owner
 	// when changefeed is deleted.
 	Cleanup(ctx context.Context) error
+
+	// Running return true if the meta manager is running or not.
+	Running() bool
```
Comment (CharlesCheung96) [inline cdc/redo/meta_manager.go:52]: Where is this interface used?

**Review by CharlesCheung96 — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**